manager/pkg/communication/marketmanager.go

```diff
@@ -26,13 +30,49 @@ type JobPayload struct {
 
 // communicate with market manager get jobs should run in this cpod.
 func GetJobs(cpodid string) []job.Job {
-	// TODO: implement it
+	// TODO: adapt marketmanager interface
+	data := map[string]string{
+		"cpod_id": cpodid,
+	}
+	bytesData, err := json.Marshal(data)
+	if err != nil {
+		return nil
+	}
+	resp, err := http.Post(baseURL+"/getjobs", "application/json", bytes.NewReader(bytesData))
+	if err != nil {
+		return nil
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		return nil
+	}
+	respData, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil
+	}
+	var parsedData map[string]interface{}
+	err = json.Unmarshal(respData, &parsedData)
+	if err != nil {
+		return nil
+	}
+	// TODO: parsedData -->  job.Job
 	return []job.Job{}
 }
 
 // upload cpod info to marketmanager.
 func UploadCPodStatus(up UploadPayload) bool {
-	// TODO: implement it
+	bytesData, err := json.Marshal(up)
+	if err != nil {
+		return false
+	}
+	resp, err := http.Post(baseURL+"/uploadstatus", "application/json", bytes.NewReader(bytesData))
+	if err != nil {
+		return false
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		return false
+	}
```

Comment:
GitHub
Actions
/ lint
S1008: should use 'return resp.StatusCode == 200' instead of 'if resp.StatusCode != 200 { return false }; return true' (gosimple)